src/main/java/org/sunbird/extension/util/TransformJsonUtil.java

```diff
@@ -0,0 +1,322 @@
+package org.sunbird.extension.util;
+
+import com.typesafe.config.Config;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class TransformJsonUtil {
+
+  private Config config;
+
+  public Map<String, Object> transform(
+      Config fieldsConfig, Map<String, Object> userInputMap, String rootConfig) {
+
+    Map<String, Object> outputMap = new HashMap<String, Object>();
+    config = fieldsConfig;
+    Set<String> userInputSet = userInputMap.keySet();
+    Iterator<String> userInputIterator = userInputSet.iterator();
+    String prefix = rootConfig + TransformationConstants.DOT;
+    while (userInputIterator.hasNext()) {
+      String key = userInputIterator.next();
+      try {
+        Map<String, Object> fieldMap = (Map<String, Object>) config.getAnyRef(prefix + key);
+        if (!fieldMap.containsKey(TransformationConstants.FROMFIELDNAME)) {
+          fieldMap.put(TransformationConstants.FROMFIELDNAME, key);
+        }
+        transformField(fieldMap, userInputMap, outputMap);
+      } catch (Exception e) {
+        // handle exception
```

Comment:
Do not suppress exception, throw to caller. Add null checks if required in try code.